src/datasource/nifti/backend.ts

```diff
@@ -295,7 +302,14 @@ export class NiftiVolumeChunkSource extends WithParameters(
       this.parameters.url,
       { signal },
     );
-    const imageBuffer = readImage(data.header, data.uncompressedData);
+    let imageBuffer = readImage(data.header, data.uncompressedData);
+    // If original header datatype was FLOAT64 (downcast) convert the buffer to Float32.
+    if (data.header.datatypeCode === NiftiDataType.FLOAT64) {
+      const src = new Float64Array(imageBuffer);
+      const dst = new Float32Array(src.length);
+      for (let i = 0; i < src.length; ++i) dst[i] = src[i];
```

Comment:
Instead of looping you can just do `Float32Array.from(src)`.  Not sure if it is any faster